### PR TITLE
feat: add xPadding

### PR DESCRIPTION
**Pull request by lpatiny**

### CODE REVIEW

File: src/x/xPadding.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * @param {Array} array - the array that will be rotated
+ * @param {object} [options={}]
+ * @param {string} [options.algorithm='value'] none, value, circular, duplicate
+ * @param {number} [options.size=0] padding size before and after
+ * @param {number} [options.value=0] value to use for padding (if algorithm='value')
+ * @return {Array}
+ */
+export function xPadding(array, options = {}) {
+  const { size = 0, value = 0, algorithm = 'value' } = options;
+  let result = new Float64Array(array.length + size * 2);
```
Comment (kjappelbaum) [inline src/x/xPadding.js:15]: Maybe we can validate the input with `import xCheck from './xCheck';`
```suggestion
  xCheck(array);
  let result = new Float64Array(array.length + size * 2);
```

**Review by kjappelbaum — CHANGES_REQUESTED** [review]
looks good to me, I think we should fix the doc string before merging.

**Comment by lpatiny** [comment]
@kjappelbaum Many thanks for your review !

**Comment by kjappelbaum** [comment]
I think one should reuse https://github.com/mljs/pad-array i just saw it and it is probably better to not maintain the code in two places.

**Pull request merged**